Tests/BitcoinCash/BitconCashTests.swift

```diff
@@ -42,4 +42,53 @@ class BitcoinCashTests: XCTestCase {
+
+        XCTAssertEqual(xprv, "xprv9yEvwSfPanK5gLYVnYvNyF2CEWJx1RsktQtKDeT6jnCnqASBiPCvFYHFSApXv39bZbF6hRaha1kWQBVhN1xjo7NHuhAn5uUfzy79TBuGiHh")
+        XCTAssertEqual(xpub, "xpub6CEHLxCHR9sNtpcxtaTPLNxvnY9SQtbcFdov22riJ7jmhxmLFvXAoLbjHSzwXwNNuxC1jUP6tsHzFV9rhW9YKELfmR9pJaKFaM8C3zMPgjw")
+    }
+
+    func testDeriveFromXPub() {
+        let xpub = "xpub6CEHLxCHR9sNtpcxtaTPLNxvnY9SQtbcFdov22riJ7jmhxmLFvXAoLbjHSzwXwNNuxC1jUP6tsHzFV9rhW9YKELfmR9pJaKFaM8C3zMPgjw"
+        let bc = BitcoinCash(purpose: .bip44)
+        let xpubAddr2 = bc.derive(from: xpub, at: bc.derivationPath(at: 2))!
+        let xpubAddr9 = bc.derive(from: xpub, at: bc.derivationPath(at: 9))!
+
+        XCTAssertEqual(xpubAddr2.description, "bitcoincash:qq4cm0hcc4trsj98v425f4ackdq7h92rsy6zzstrgy")
+        XCTAssertEqual(xpubAddr9.description, "bitcoincash:qqyqupaugd7mycyr87j899u02exc6t2tcg9frrqnve")
+    }
+
+    func testSignTransaction() throws {
+        // Transaction on Bitcoin Cash Mainnet
+        // https://blockchair.com/bitcoin-cash/transaction/96ee20002b34e468f9d3c5ee54f6a8ddaa61c118889c4f35395c2cd93ba5bbb4
+        let toAddress = BitcoinAddress(string: "1Bp9U1ogV3A14FMvKbRJms7ctyso4Z4Tcx")!
+        let changeAddress = BitcoinAddress(string: "1FQc5LdgGHMHEN9nwkjmz6tWkxhPpxBvBU")!
+
+        let unspentOutput = BitcoinTransactionOutput(value: 5151, script: BitcoinScript(data: Data(hexString: "76a914aff1e0789e5fe316b729577665aa0a04d5b0f8c788ac")!))
+
+        let unspentOutpoint = BitcoinOutPoint(hash: Data(hexString: "e28c2b955293159898e34c6840d99bf4d390e2ee1c6f606939f18ee1e2000d05")!, index: 2)
+        let utxo = BitcoinUnspentTransaction(output: unspentOutput, outpoint: unspentOutpoint)
+        let utxoKey = PrivateKey(wif: "L1WFAgk5LxC5NLfuTeADvJ5nm3ooV3cKei5Yi9LJ8ENDfGMBZjdW")!
+
+        let provider = BitcoinDefaultPrivateKeyProvider(keys: [utxoKey])
+
+        let amount: Int64 = 600
+
+        let unsignedTx = try BitcoinTransaction.build(to: toAddress, amount: amount, fee: 226, changeAddress: changeAddress, utxos: [utxo])
+
+        let signer = BitcoinTransactionSigner(keyProvider: provider, transaction: unsignedTx, hashType: [.fork, .all])
+        let signedTx = try signer.sign([utxo])
+
+        var serialized = Data()
+        signedTx.encode(into: &serialized)
+
+        XCTAssertEqual(signedTx.transactionId, "96ee20002b34e468f9d3c5ee54f6a8ddaa61c118889c4f35395c2cd93ba5bbb4")
+        // swiftlint:disable:next line_length
+        XCTAssertEqual(serialized.hexString, "0100000001e28c2b955293159898e34c6840d99bf4d390e2ee1c6f606939f18ee1e2000d05020000006b483045022100b70d158b43cbcded60e6977e93f9a84966bc0cec6f2dfd1463d1223a90563f0d02207548d081069de570a494d0967ba388ff02641d91cadb060587ead95a98d4e3534121038eab72ec78e639d02758e7860cdec018b49498c307791f785aa3019622f4ea5bffffffff0258020000000000001976a914769bdff96a02f9135a1d19b749db6a78fe07dc9088ace5100000000000001976a9149e089b6889e032d46e3b915a3392edfd616fb1c488ac00000000")
```

Comment:
very weird, though we have passed this case, but new tx still gets rejected: {"error":"-26: 16: mandatory-script-verify-flag-failed (Signature must be zero for failed CHECK(MULTI)SIG operation)"}

Reply:
@alejandro-isaza could you please help this?

Reply:
The signature is correct. Maybe there is something wrong with the transaction you are sending. How are you testing that?

Reply:
it looks like an issue on our bcash nodee and trezor bitcoin abc node, I print the raw tx and send it to an electrum x server, and works